File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/573_:::_The_Snail_★/Main.cpp
using namespace std;
#include <cstdio>
#include <iostream>
#include <string>

int main()
{
	double i, posSnail;
	int H, U, D, F;
	string r1;

	cin >> H >> U >> D >> F;
	while(H!=0 && U!=0 && D!=0 && F!=0)
	{
		i=1; posSnail=0;
		while(true)
		{
			double subida=(U-(U*(i-1)*F/100));
			posSnail+=(subida>=0)?subida:0;
			if(posSnail>H)break;
			posSnail-=D;
			if(posSnail<0)break;
			i++;
		}

		r1=(posSnail<0)?"failure":"success";
		cout << r1 << " on day " << (int)i << endl;

		cin >> H >> U >> D >> F;
	}

	return 0;
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/573_:::_The_Snail_★/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		int H, U, D, F;
		double i, posSnail, subida;

		for(String linea, r1, param[]; !(linea=br.readLine()).equals("0 0 0 0");)
		{
			param=linea.split(" ");
			H=Integer.parseInt(param[0]);
			U=Integer.parseInt(param[1]);
			D=Integer.parseInt(param[2]);
			F=Integer.parseInt(param[3]);

			i=1;
			posSnail=0;

			while(true)
			{
				subida=(U-(U*(i-1)*F/100));
				posSnail+=(subida>=0)?subida:0;
				if(posSnail>H)break;
				posSnail-=D;
				if(posSnail<0)break;
				i++;
			}

			r1=(posSnail<0)?"failure":"success";
			System.out.println(r1+" on day "+(int)i);
		}
		//-----------------------------------------------------------
		br.close();
	}
}
